lib/Service/ShareService.php

```diff
@@ -0,0 +1,112 @@
+<?php
+/**
+ * Nextcloud - News
+ *
+ * This file is licensed under the Affero General Public License version 3 or
+ * later. See the COPYING file.
+ *
+ * @author    Marco Nassabain <marco.nassabain@hotmail.com>
+ */
+
+namespace OCA\News\Service;
+
+use \OCA\News\Db\Item;
+use \OCA\News\Db\Feed;
+
+use \Psr\Log\LoggerInterface;
+
+use OCA\News\Service\Exceptions\ServiceNotFoundException;
+use OCP\AppFramework\Db\DoesNotExistException;
+
+/**
+ * Class ShareService
+ *
+ * @package OCA\News\Service
+ */
+class ShareService
+{
+    /**
+     * Items service.
+     *
+     * @var ItemServiceV2
+     */
+    protected $itemService;
+
+    /**
+     * Feeds service.
+     *
+     * @var FeedServiceV2
+     */
+    protected $feedService;
+    
+    /**
+     * @var LoggerInterface
+     */
+    protected $logger;
+
+    /**
+     * ShareService constructor
+     *
+     * @param FeedServiceV2   $feedService Service for feeds
+     * @param ItemServiceV2   $itemService Service to manage items
+     * @param LoggerInterface $logger      Logger
+     */
+    public function __construct(
+        FeedServiceV2 $feedService,
+        ItemServiceV2 $itemService,
+        LoggerInterface $logger
+    ) {
+        $this->itemService = $itemService;
+        $this->feedService = $feedService;
+        $this->logger      = $logger;
+    }
+
+    /**
+     * Share an item with a user
+     *
+     * @param string $userId           ID of user sharing the item
+     * @param int    $itemId           Item ID
+     * @param string $shareRecipientId ID of user to share with
+     *
+     * Sharing by copying - the item is duplicated, and the 'sharedBy'
+     * field is filled accordingly.
+     * The item is then placed in a dummy feed reserved for items
+     * shared with the user
+     *
+     * @return Item Shared item
+     * @throws ServiceNotFoundException|ServiceConflictException
+     */
+    public function shareItemWithUser(string $userId, int $itemId, string $shareRecipientId)
+    {
+        // Find item to share
+        $item = $this->itemService->find($userId, $itemId);
+
+        // Duplicate item & initialize fields
+        $sharedItem = clone $item;
+        $sharedItem->setId(null);
+        $sharedItem->setUnread(true);
+        $sharedItem->setStarred(false);
+        $sharedItem->setSharedBy($userId);
+
+        // Get 'shared with me' dummy feed
+        $feedUrl = 'http://nextcloud/sharedwithme';
```

Comment:
I'm not sure how well a complete dummy url works, but the import uses the same pattern (`http://nextcloud/nofeed`).
Maybe something standardized like `urn:nextcloud:sharedwithme` would work too but I'm not sure whether clients expect an actual http(s) url.

Reply:
On that case I'd go for `https://serverurl/news/shared` or something. Arguably we can get away with a lot more magic on imports because the user initiates it.